src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -111,115 +111,71 @@ public DataWriter(
       Time time
   ) {
     this.time = time;
-    try {
-      System.setProperty("hadoop.home.dir", config.hadoopHome());
-
-      this.connectorConfig = config;
-      this.avroData = avroData;
-      this.context = context;
-
-      log.info("Hadoop configuration directory {}", config.hadoopConfDir());
-      Configuration conf = config.getHadoopConfiguration();
-      if (!config.hadoopConfDir().equals("")) {
-        conf.addResource(new Path(config.hadoopConfDir() + "/core-site.xml"));
-        conf.addResource(new Path(config.hadoopConfDir() + "/hdfs-site.xml"));
-      }
-
-      if (config.kerberosAuthentication()) {
-        SecurityUtil.setAuthenticationMethod(
-            UserGroupInformation.AuthenticationMethod.KERBEROS,
-            conf
-        );
+    this.connectorConfig = config;
+    this.avroData = avroData;
+    this.context = context;
+    url = config.url();
+    topicDirs = new HashMap<>();
+    logDirs = new HashMap<>();
+    hiveIntegration = config.getBoolean(HiveConfig.HIVE_INTEGRATION_CONFIG);
```

Comment:
if we are going to go down this path, let's add config getters and remove all variables that gotten from configs i.e `url` and `hiveIntegration`

Reply:
sure, sounds good